reference.py

```diff
@@ -61,9 +61,26 @@ def get_pubkey_from_input(vin: VinInfo) -> ECPubKey:
         if (pubkey.valid) & (pubkey.compressed):
             return pubkey
     if is_p2tr(vin.prevout):
+        witnessStack = vin.txinwitness.scriptWitness.stack
+        if (len(witnessStack) > 1):
+            # Script spend
+
+            if (len(witnessStack) > 2 and witnessStack[-1][0] == 0x50):
+                # Last item is annex
+                witnessStack.pop()
+            
+            control_block = witnessStack[-1]
+            #  control block is <control byte> <32 byte internal key> and 1 or more <32 byte hash>
+            internal_key = control_block[1:33]
+            if (internal_key == NUMS_H.to_bytes(32, 'big')):
+                # Skip if NUMS_H
+                return ECPubKey()
```

Comment:
better `return None`?

Reply:
`get_pubkey_from_input` already returned this invalid pubkey as the default response. Users of the function do a `pubkey.valid` check before using it. We can change it to return `None`. I don't know which is better. I'm fine with either way.

Reply:
I think None is clearer, but that's just a nit, feel free to do whatever you like better